WorkflowTesting/Sources/RenderExpectations.swift

```diff
@@ -123,15 +124,44 @@ extension ExpectedSideEffect {
 }
 
 public struct ExpectedWorkflow {
-    let workflowType: Any.Type
-    let key: String
     let rendering: Any
     let output: Any?
+    let doesMatch: (_ childWorkflow: Any, _ childKey: String) -> Bool
```

Comment:
This sounds strange to me. Can we call this `matches` instead?

Reply:
Closing this one, in favor of #15